tests/Microsoft.Identity.Web.Test/L1L2CacheTests.cs

```diff
@@ -51,6 +52,30 @@ public async Task WriteCache_WritesInL1L2_TestAsync()
             Assert.Single(L2Cache.dict);
         }
 
+        [Fact]
+        public async Task WriteCache_NegativeExpiry_TestAsync()
+        {
+            // Arrange
+            byte[] cache = new byte[3];
+            AssertCacheValues(_testCacheAdapter);
+            Assert.Equal(0, _testCacheAdapter._memoryCache.Count);
+            Assert.Empty(L2Cache.dict);
+            CacheSerializerHints cacheSerializerHints = new CacheSerializerHints();
+            cacheSerializerHints.SuggestedCacheExpiry = System.DateTimeOffset.Now - System.TimeSpan.FromHours(1);
+
+            // Act
+            await _testCacheAdapter.TestWriteCacheBytesAsync(DefaultCacheKey, cache, cacheSerializerHints).ConfigureAwait(false);
+
+            // Assert
+            Assert.Equal(1, _testCacheAdapter._memoryCache.Count);
+            Assert.Single(L2Cache.dict);
+            await Task.Delay(1000).ConfigureAwait(false);
```

Comment:
Generally we should avoid awaiting in tests. But I think in this case the adapter should just delete the entry itself or not set an expiry.

Reply:
what happens if MSAL expects that the entry was written and doesn't find it just after? I tried to have a void implementation of a serializer and that didn't work. MSAL crashed, it was expecting something to come back.